### PR TITLE
fix(gesture): don't trigger swipe if not the main direction

### PR DESCRIPTION
**Pull request by code-tree**
A horizontal swipe action should not be triggered if the main direction of
the swipe was actually vertical (and vice versa).

#7911

### CODE REVIEW

**Comment by DevVersion** [comment]
The logic is valid in my opinion. So LGTM. - Needs to be manually tested.

File: src/core/services/gesture/gesture.js [hunk 1]
```diff
@@ -305,15 +305,18 @@ function MdGesture($$MdGestureHandler, $$rAF, $timeout) {
...
-          Math.abs(pointer.distanceX) > this.state.options.minDistance) {
-          eventType = pointer.directionX == 'left' ? '$md.swipeleft' : '$md.swiperight';
-          this.dispatchEvent(ev, eventType);
-        }
-        else if (Math.abs(pointer.velocityY) > this.state.options.minVelocity &&
-          Math.abs(pointer.distanceY) > this.state.options.minDistance) {
-          eventType = pointer.directionY == 'up' ? '$md.swipeup' : '$md.swipedown';
-          this.dispatchEvent(ev, eventType);
+        if (Math.abs(pointer.distanceX) > Math.abs(pointer.distanceY)) {
+            if (Math.abs(pointer.velocityX) > this.state.options.minVelocity &&
```
Comment (ThomasBurleson) [inline src/core/services/gesture/gesture.js]: Please refactor this into a distinct function call. eg.

```
if ( isHorizontalSwipe(pointer, this.state.options) ) {

}
```

and `isVerticalSwipe(pointer, this.state.options)`

Comment (switcharray) [inline src/core/services/gesture/gesture.js]: jtyfj

File: src/core/services/gesture/gesture.js [hunk 2]
```diff
@@ -305,18 +305,26 @@ function MdGesture($$MdGestureHandler, $$rAF, $timeout) {
...
-            }
+        if (isHorizontalSwipe(pointer, this.state.options)) {
+          eventType = pointer.directionX == 'left' ? '$md.swipeleft' : '$md.swiperight';
+          this.dispatchEvent(ev, eventType);
+        } else if (isVerticalSwipe(pointer, this.state.options)) {
+          eventType = pointer.directionY == 'up' ? '$md.swipeup' : '$md.swipedown';
+          this.dispatchEvent(ev, eventType);
+        }
+        
+        function isHorizontalSwipe(pointer, options) {
```
Comment (code-tree) [inline src/core/services/gesture/gesture.js]: @ThomasBurleson how does this look? Any further changes needed?

**Comment by code-tree** [comment]
I suppose tests could also be added for this in spec.js?

**Comment by ThomasBurleson** [comment]
This issue is closed as part of our ‘Surge Focus on Material 2' efforts.
      For details, see our forum posting @ http://bit.ly/1UhZyWs.

**Pull request closed without merging**